File: backend/classes/Company.php
<?php
class Company {
    private $conn;
    private $table_name = "companies";

    public $id;
    public $name;
    public $email;
    public $phone;
    public $address;
    public $tax_id;
    public $logo_path;
    public $currency;
    public $default_tax_rate;
    public $invoice_prefix;
    public $created_at;
    public $updated_at;

    public function __construct($db) {
        $this->conn = $db;
    }

    public function create() {
        $query = "INSERT INTO " . $this->table_name . " 
                SET name=:name, email=:email, phone=:phone, 
                    address=:address, tax_id=:tax_id, logo_url=:logo_url,
                    currency=:currency, default_tax_rate=:default_tax_rate, 
                    invoice_prefix=:invoice_prefix";

        $stmt = $this->conn->prepare($query);

        $this->name = htmlspecialchars(strip_tags($this->name));
        $this->email = htmlspecialchars(strip_tags($this->email));
        $this->phone = htmlspecialchars(strip_tags($this->phone));
        $this->address = htmlspecialchars(strip_tags($this->address));
        $this->tax_id = htmlspecialchars(strip_tags($this->tax_id));
        $this->logo_path = htmlspecialchars(strip_tags($this->logo_path));
        $this->currency = htmlspecialchars(strip_tags($this->currency));
        $this->default_tax_rate = htmlspecialchars(strip_tags($this->default_tax_rate));
        $this->invoice_prefix = htmlspecialchars(strip_tags($this->invoice_prefix));

        $stmt->bindParam(":name", $this->name);
        $stmt->bindParam(":email", $this->email);
        $stmt->bindParam(":phone", $this->phone);
        $stmt->bindParam(":address", $this->address);
        $stmt->bindParam(":tax_id", $this->tax_id);
        $stmt->bindParam(":logo_url", $this->logo_path);
        $stmt->bindParam(":currency", $this->currency);
        $stmt->bindParam(":default_tax_rate", $this->default_tax_rate);
        $stmt->bindParam(":invoice_prefix", $this->invoice_prefix);

        if($stmt->execute()) {
            $this->id = $this->conn->lastInsertId();
            return true;
        }
        return false;
    }

    public function read() {
        $query = "SELECT * FROM " . $this->table_name . " LIMIT 1";
        $stmt = $this->conn->prepare($query);
        $stmt->execute();
        
        return $stmt;
    }

    public function update() {
        $query = "UPDATE " . $this->table_name . " 
                SET name=:name, email=:email, phone=:phone, 
                    address=:address, tax_id=:tax_id, logo_path=:logo_path,
                    currency=:currency, default_tax_rate=:default_tax_rate, 
                    invoice_prefix=:invoice_prefix
                WHERE id=:id";

        $stmt = $this->conn->prepare($query);

        $this->name = htmlspecialchars(strip_tags($this->name));
        $this->email = htmlspecialchars(strip_tags($this->email));
        $this->phone = htmlspecialchars(strip_tags($this->phone));
        $this->address = htmlspecialchars(strip_tags($this->address));
        $this->tax_id = htmlspecialchars(strip_tags($this->tax_id));
        $this->logo_path = htmlspecialchars(strip_tags($this->logo_path));
        $this->currency = htmlspecialchars(strip_tags($this->currency));
        $this->default_tax_rate = htmlspecialchars(strip_tags($this->default_tax_rate));
        $this->invoice_prefix = htmlspecialchars(strip_tags($this->invoice_prefix));
        $this->id = htmlspecialchars(strip_tags($this->id));

        $stmt->bindParam(":name", $this->name);
        $stmt->bindParam(":email", $this->email);
        $stmt->bindParam(":phone", $this->phone);
        $stmt->bindParam(":address", $this->address);
        $stmt->bindParam(":tax_id", $this->tax_id);
        $stmt->bindParam(":logo_path", $this->logo_path);
        $stmt->bindParam(":currency", $this->currency);
        $stmt->bindParam(":default_tax_rate", $this->default_tax_rate);
        $stmt->bindParam(":invoice_prefix", $this->invoice_prefix);
        $stmt->bindParam(":id", $this->id);

        if($stmt->execute()) {
            return true;
        }
        return false;
    }
}
?>

File: lib/models/company.dart
class Company {
  final String id;
  final String name;
  final String email;
  final String phone;
  final String address;
  final String taxId;
  final String? logoPath;
  final String currency;
  final double defaultTaxRate;
  final String invoicePrefix;
  final DateTime createdAt;
  final DateTime updatedAt;

  Company({
    required this.id,
    required this.name,
    required this.email,
    required this.phone,
    required this.address,
    required this.taxId,
    this.logoPath,
    this.currency = 'USD',
    this.defaultTaxRate = 0.0,
    this.invoicePrefix = 'INV',
    required this.createdAt,
    required this.updatedAt,
  });

  Map<String, dynamic> toMap() {
    return {
      'id': id,
      'name': name,
      'email': email,
      'phone': phone,
      'address': address,
      'taxId': taxId,
      'logoPath': logoPath,
      'currency': currency,
      'defaultTaxRate': defaultTaxRate,
      'invoicePrefix': invoicePrefix,
      'createdAt': createdAt.millisecondsSinceEpoch,
      'updatedAt': updatedAt.millisecondsSinceEpoch,
    };
  }

  factory Company.fromMap(Map<String, dynamic> map) {
    return Company(
      id: map['id']?.toString() ?? '',
      name: map['name']?.toString() ?? '',
      email: map['email']?.toString() ?? '',
      phone: map['phone']?.toString() ?? '',
      address: map['address']?.toString() ?? '',
      taxId: map['taxId']?.toString() ?? '',
      logoPath: map['logoPath']?.toString(),
      currency: map['currency']?.toString() ?? 'USD',
      defaultTaxRate: map['defaultTaxRate']?.toDouble() ?? 0.0,
      invoicePrefix: map['invoicePrefix']?.toString() ?? 'INV',
      createdAt: map['createdAt'] != null 
          ? DateTime.fromMillisecondsSinceEpoch(map['createdAt']) 
          : DateTime.now(),
      updatedAt: map['updatedAt'] != null 
          ? DateTime.fromMillisecondsSinceEpoch(map['updatedAt']) 
          : DateTime.now(),
    );
  }

  Company copyWith({
    String? name,
    String? email,
    String? phone,
    String? address,
    String? taxId,
    String? logoPath,
    String? currency,
    double? defaultTaxRate,
    String? invoicePrefix,
    DateTime? updatedAt,
  }) {
    return Company(
      id: id,
      name: name ?? this.name,
      email: email ?? this.email,
      phone: phone ?? this.phone,
      address: address ?? this.address,
      taxId: taxId ?? this.taxId,
      logoPath: logoPath ?? this.logoPath,
      currency: currency ?? this.currency,
      defaultTaxRate: defaultTaxRate ?? this.defaultTaxRate,
      invoicePrefix: invoicePrefix ?? this.invoicePrefix,
      createdAt: createdAt,
      updatedAt: updatedAt ?? DateTime.now(),
    );
  }
}
